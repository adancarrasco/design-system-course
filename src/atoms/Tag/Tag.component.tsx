import * as React from 'react'
import styled from '@emotion/styled'

export interface ITagProps extends ITagStyledProps {
  backgroundColor?: string
}

const Tag: React.FC<ITagProps> = ({ ...rest }) => <TagStyled {...rest} />

export interface ITagStyledProps {
  backgroundColor?: string
  backgroundColorHover?: string
  isExpanded?: boolean
}

const TagStyled = styled.span<ITagStyledProps>`
  background-color: ${(props) => props.backgroundColor ?? '#ababab'};
  cursor: pointer;
  display: inline-flex;
  width: 3rem;
  height: ${(props) => (props.isExpanded ? '2.5rem' : '0.4rem')};
  border-radius: 3px;
  :hover {
    background-color: ${(props) => props.backgroundColorHover ?? '#ebebeb'};
  }
`

export default Tag
