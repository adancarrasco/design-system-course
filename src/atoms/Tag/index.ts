export { default } from './Tag.component'
export type { ITagProps } from './Tag.component'
