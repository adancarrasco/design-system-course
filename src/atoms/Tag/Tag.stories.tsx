import React from 'react'
import { storiesOf } from '@storybook/react'

import Tag from '.'

storiesOf('Atoms|Tag', module)
  .add('Default', () => <Tag />)
  .add('Custom props', () => <Tag backgroundColor="#90ce89" />)
  .add('As expanded', () => <Tag isExpanded />)
