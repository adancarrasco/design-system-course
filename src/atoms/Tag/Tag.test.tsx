import React from 'react'
import { render, getByTestId } from '@testing-library/react'
import { matchers } from 'jest-emotion'
import '@testing-library/jest-dom/extend-expect'

import Tag, { ITagProps } from '.'

expect.extend(matchers)

describe('Tag Component', () => {
  const renderComponent = (props: ITagProps, dataTestId: string) =>
    render(<Tag data-testid={dataTestId} {...props} />)

  it('matches snapshot and default render', () => {
    const dataTestId = 'tag'
    const props = {}
    const { getByTestId } = renderComponent(props, dataTestId)
    const container = getByTestId('tag')
    expect(container).toMatchSnapshot()
    expect(container).toBeInTheDocument()
  })

  it('should render as expanded', () => {
    const dataTestId = 'tag'
    const props = {
      isExpanded: true,
    }
    const { getByTestId } = renderComponent(props, dataTestId)
    const container = getByTestId('tag')
    expect(container).toHaveStyleRule('height', '2.5rem')
  })

  it('should render with custom props', () => {
    const dataTestId = 'tag'
    const props = {
      backgroundColor: 'green',
    }
    const { getByTestId } = renderComponent(props, dataTestId)
    const container = getByTestId('tag')
    expect(container).toHaveStyleRule('background-color', 'green')
  })
})
