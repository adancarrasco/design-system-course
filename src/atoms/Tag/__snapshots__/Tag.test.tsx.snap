// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag Component matches snapshot and default render 1`] = `
.emotion-0 {
  background-color: #ababab;
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 3rem;
  height: 0.4rem;
  border-radius: 3px;
}

.emotion-0:hover {
  background-color: #ebebeb;
}

<span
  class="emotion-0"
  data-testid="tag"
/>
`;
